packages/sitecore-jss-dev-tools/src/templating/utils.ts

```diff
@@ -0,0 +1,222 @@
+    encoding: 'utf8',
+  });
+}
+
+/**
+ * @param {string} path path to get plugin from
+ * @param {string} pluginName plugin name
+ */
+export function getPluginList(path: string, pluginName: string): PluginFile[] {
+  const plugins = getItems<PluginFile>({
+    path,
+    resolveItem: (path, name) => ({
+      path: `${path}/${name}`,
+      name: `${name.replace(/-./g, (x) => x[1].toUpperCase())}Plugin`,
+    }),
+    cb: (name) => console.debug(`Registering ${pluginName} plugin ${name}`),
+  });
+
+  return plugins;
+}
+
+/**
+ * Creates a file relative to the specified path if the file doesn't exist. Creates directories as needed.
+ * @param {string} rootPath - the root path
+ * @param {string} fileContent - the file content
+ * @param {string} filename - the filename
+ * @param {string} componentPath - path to put components into
+ * @returns the new file's filepath
+ */
+export function scaffoldFile(
+  rootPath: string,
+  fileContent: string,
+  filename: string,
+  componentPath: string
+): string | null {
+  const outputDir = path.join(rootPath, componentPath);
+  const outputFile = path.join(outputDir, filename);
+
+  if (fs.existsSync(outputFile)) {
+    console.log(chalk.red(`Skipping creating ${outputFile}; already exists.`));
+    return null;
+  }
+
+  fs.mkdirSync(outputDir, { recursive: true });
+  fs.writeFileSync(outputFile, editLineEndings(fileContent), 'utf8');
+  console.log(chalk.green(`File ${outputFile} has been scaffolded.`));
+  return outputFile;
+}
+
+/**
+ * Run watch mode, watching on @var paths
+ * @param {string[]} paths paths to watch by chokidar
+ * @param {Function<void>}componentFactoryCallback component factory method to call on changes to watched components
+ */
+export function watchItems(paths: string[], componentFactoryCallback: () => void): void {
+  chokidar
+    .watch(paths, { ignoreInitial: true, awaitWriteFinish: true })
+    .on('add', componentFactoryCallback)
+    .on('unlink', componentFactoryCallback);
+}
+
+/**
+ * Force to use `crlf` line endings, we are using `crlf` across the project.
+ * Replace: `lf` (\n), `cr` (\r)
+ * @param {string} content
+ */
+export function editLineEndings(content: string) {
+  return content.replace(/\r|\n/gm, '\r\n');
+}
+
+/**
+ * @param {string} path plugins path
+ * @param resolveItem will resolve item in required data format
+ * @param cb will be called when new item is found
+ * @param {RegExp} fileFormat Matches specific files
+ * @param {boolean} recursive if true will search recursively
+ */
+type GetItemsSettings<Item> = {
+  path: string;
+  resolveItem: (path: string, name: string) => Item;
+  cb?: (name: string) => void;
+  fileFormat?: RegExp;
+  recursive?: boolean;
+};
+
+/**
+ * Type to specify plugin file details
+ */
+interface PluginFile {
+  path: string;
+  name: string;
+}
+
+/**
+ * Definition to be used for plugin registration during bootstrap
+ * listPath - destination path to compile plugins to
```

Comment:
If you decided to describe parameters, in this case comments needs to be added by general convention
```
export interface PluginDefinition {
   /*
   * comment
  */
   listPath: string;
   /*
   * comment
  */
   rootPath: string;
}
```